components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1463,6 +1474,60 @@ def get_thumb_url(iid):
 
 @login_required()
 @jsonp
+def get_thumbnails_json(request, w=None, conn=None, **kwargs):
+    """
+    Returns base64 encoded jpeg with the rendered thumbnail for images
+    'id'
+
+    @param request:     http request
+    @param w:           Thumbnail max width. 96 by default
+    @return:            http response containing base64 encoded thumbnails
+    """
+    if w is None:
+        w = 96
+    image_ids = get_longs(request, 'id')
+    logger.debug("Image ids: %r" % image_ids)
+    if len(image_ids) > settings.THUMBNAILS_BATCH:
+        return HttpJavascriptResponseServerError(
+            'Max %s thumbnails at a time.' % settings.THUMBNAILS_BATCH)
```

Comment:
Is the primary reason for the THUMBNAILS_BATCH setting to limit the size of the request string?
If so, then by the time we've got to this point, we know that the user has not exceeded this limit.
So is there any reason to reject the request at this stage?
Maybe the behaviour should be more like PAGE_SIZE, which is a default for the browser code to use, but the python code will still support larger page sizes if requested.